app/src/main/java/com/andryoga/safebox/ui/view/chooseMasterPswrd/ChooseMasterPswrdFragment.kt

```diff
@@ -37,12 +44,46 @@ class ChooseMasterPswrdFragment : Fragment() {
         binding.viewModel = viewModel
         binding.lifecycleOwner = this
 
+        binding.hintText.addTextChangedListener {
+            val similarLength = longestCommonSubstring(
+                binding.hintText.text.toString(),
+                binding.pswrdText.text.toString()
+            )
+
+            if (similarLength >= Constants.maxSimilarLength) {
+                binding.hint.isErrorEnabled = true
```

Comment:
few issues:
1. first enter 5 consecutive letters, error will be shown... now change hint to correct hint.. error is still shown. Ideally as soon as hint validation pass, error should be cleared
2. match should not be case sensitive. e.g. enter password as Qwert... enter hint as qwert... error is **not** shown. we should show error here as well.
3. even if validation fails, we are still able to click on save which will store this hint in db.
4. see [this](https://material.io/components/text-fields/android#using-text-fields) to learn how you can make hint length more user friendly.. search for **counter**

Reply:
Yes I am working on (1) and I will take care of the rest